algosdk/future/transaction.py

```diff
@@ -434,47 +438,148 @@ def dictify(self):
 
         return od
 
+    def __eq__(self, other):
+        if not isinstance(other, (
+                KeyregTxn,
+                transaction.KeyregTxn)):
+            return False
+        return (super(KeyregTxn, self).__eq__(other) and
+                self.votepk == other.votepk and
+                self.selkey == other.selkey and
+                self.votefst == other.votefst and
+                self.votelst == other.votelst and
+                self.votekd == other.votekd)
```

Comment:
I think this should also compare `nonpart`.

Reply:
fixed.